### DISCUSSION TITLE
python script can't run through deta.sh

### DISCUSSION THREAD

**Discussion by a5kgly** [deta/cloud-docs#488] [Help] [answered]
Hi there,

I have the python script(main.py) run fine in the local machine.

But it doesn't work through deta.sh with cron set to run everyday.

deta cron set "10 16 * * ? *"

Below is the error log from deta.sh.

Traceback (most recent call last):
  File "/opt/python/detalib/debugger.py", line 142, in wrap
    result = func(event, context)
  File "/var/task/_entry.py", line 14, in handler
    return handle(event, main)
  File "/opt/python/detalib/handler.py", line 8, in handle
    assert hasattr(
AssertionError: We couldn't find your 'app'. Please refer to the Manual.

Could someone please help me to figure out how to fix it?

Thanks a lot.

**Comment by jnsougata** [comment]
In the root of your micro directory there should be a `main.py` file inside which you should have an identifier strictly named `app`  which instantiates your ASGI or WSGI application 

for example: `app = FastAPI()`

**Reply by Sokhavuth** [reply]
Yes, Deta platform requires Python application to have the main file be named "main.py" in which the main application be named "app". For the rest of files, we could name them anything we like.

**Comment by abdelhai** [comment, marked as answer]
please follow the cron example here: https://docs.deta.sh/docs/micros/cron
make sure you import `app` from `deta` and use it
